llvm/utils/TableGen/DAGISelEmitter.cpp

```diff
@@ -132,7 +132,8 @@ struct PatternSortingPredicate {
 } // End anonymous namespace
 
 void DAGISelEmitter::run(raw_ostream &OS) {
-  Records.startTimer("Parse patterns");
+  RecordKeeper &MutableRC = const_cast<RecordKeeper &>(Records);
```

Comment:
I can see that you are making it `const` in the class and then removing it with `const_cast` so that `startTimer()` works. However I think `const_cast` is hacky and should be avoided unless necessary, and here it's only introduced because of the change you are making.
- Can the timing functionality be separated from `RecordKeeper`?
- Can `startTimer` be made `const` and somehow hide the `const_cast` inside `RecordKeeper`?
- What do we actually gain by making `RecordKeeper &Records` `const`?

Reply:
I have addressed this now and removed the const_cast here.